### DISCUSSION TITLE
run `python -m pretalx rebuild` without internet access?

### DISCUSSION THREAD

**Discussion by jmrohwer** [pretalx/pretalx#1693] [Self-hosting] [answered]
I'm running a self-hosted instance of pretalx on a server that is located in the DMZ firewall zone of our University. This means that the server cannot initiate any internet connections to the outside (either campus or rest of the internet), and all connections have to be initiated **from the outside to the server**. The only exception is the university's own FTP archive, which for the purposes of this discussion means that the server can only access a mirror of the Ubuntu repos.

I've gotten around the issue of installing the python packages by using `pip download` on a separate machine running the same version of Python, copying all the files over to the server, and then using `pip install --no-index --find-links /directory/where/wheels/are/stored --user -U ...` on the server itself.

All works until this point but when I try to run `python -m pretalx rebuild` it first copies the static files (that works okay) but then hangs on the vite/npm build because there is no internet access. I have gotten around this by building pretalx on a virtual machine on my laptop (again with the same OS as the server), and then recursively copying the content of `/var/pretalx/static` to the server. This seems to work, however it is very cumbersome, especially when wanting to do upgrades.

Is there a **better way**? For example, could the required files be downloaded separately and then copied over to the server in a similar way as the Python wheels? Thanks for your help.

**Comment by rixx** [comment, marked as answer]
I'm afraid running pretalx without internet access is currently out of scope.

The `rebuild` command checks for the presence of the `node_modules` directory in order to decide if `npm install` should be run, which requires internet access. Building pretalx packages / wheels with the full `node_modules` directory included would result in a fairly large bundle, and setting this build up to work reliably is currently just not a priority, I'm afraid – we did consider it for a bit, but as we'd still require nodejs to be installed on the server, we judged it to be not worth the effort it would require.
